### PR TITLE
manifest: openthread: Simplification of openthread integration.

### PR DESCRIPTION
**Pull request by canisLupus1313**
This commit moves openthread CMakeLists.txt from openthread submodule
to zephyr/modules/openthread.

Signed-off-by: Przemyslaw Bida <przemyslaw.bida@nordicsemi.no>

### CODE REVIEW

**Review by rlubos — DISMISSED** [review]

File: subsys/net/l2/openthread/Kconfig.features [hunk 1]
```diff
@@ -208,3 +208,12 @@ config OPENTHREAD_SRP_CLIENT
...
 	select OPENTHREAD_ECDSA
+
+config OPENTHREAD_CSL_DEBUG
+	bool "Enable CSL debugging"
+
+config OPENTHREAD_DATSET_UPDATER
+	bool "Enable dataset updater"
+
+config OPENTHREAD_UPTIME
+	bool "Enable openthread uptime counter."
```
Comment (edmont) [inline subsys/net/l2/openthread/Kconfig.features:219]:
```suggestion
	bool "Enable openthread uptime counter"
```

**Review by edmont — DISMISSED** [review]

File: subsys/net/l2/openthread/Kconfig.features (hunk 1 above)
Comment (canisLupus1313) [reply subsys/net/l2/openthread/Kconfig.features]: fixed

**Review by edmont — DISMISSED** [review]

File: west.yml [hunk 2]
```diff
@@ -195,7 +195,7 @@ manifest:
       revision: cfd050ff38a9d028dc211690b2ec35971128e45e
       path: modules/lib/open-amp
     - name: openthread
-      revision: 4ae8fe2ef1bfe0a96188d3dfa970f5d63386281b
+      revision: pull/61/head
```
Comment (rlubos) [inline west.yml:198]: Please update SHA: `0e141a556a11d82b1ee7fda00be9d1553375a687`

**Review by rlubos — APPROVED** [review]

**Review by edmont — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

File: subsys/net/l2/openthread/Kconfig.features [hunk 3]
```diff
@@ -208,3 +208,12 @@ config OPENTHREAD_SRP_CLIENT
 config OPENTHREAD_SRP_SERVER
 	bool "Enable SRP Server support"
 	select OPENTHREAD_ECDSA
+
+config OPENTHREAD_CSL_DEBUG
+	bool "Enable CSL debugging"
+
+config OPENTHREAD_DATSET_UPDATER
```
Comment (pdgendt) [inline subsys/net/l2/openthread/Kconfig.features:215]: A bit late to the party, but there is a typo here.

**Pull request merged**